enhancements/cluster-logging/cluster-logging-es-rollover-data-design.md

```diff
@@ -0,0 +1,253 @@
+
+#### ClusterLogging API
+```
+apiVersion: "logging.openshift.io/v1"
+kind: "ClusterLogging"
+metadata:
+  name: "instance"
+spec:
+  logStore:
+    retentionPolicy: 
+      logs.app:
+        maxAge: 7d
+      logs.infra
+        maxAge: 7d
+
+```
+#### Cluster Logging Operator
+The `cluster-logging-operator` will utilized the ClusterLogging CR retention policy to spec
+the desired aforementioned Elasticsearch CR indexManagement.
+#### Elasticsearch API
+```
+apiVersion: "logging.openshift.io/v1"
+kind: "Elasticsearch"
+metadata:
+  name: "elasticsearch"
+spec:
+  indexManagement:
+    policies:
+    - name: infra-policy
+      phases:
+        hot:
+          actions:
+            rollover:
+              max_size:  30GB         #opinionated value on Elastic recommendations
+              max_age:   3d
+        delete:
+          min_age: 7d
+    templates:
+    - name:  infra
+      template:
+        index_patterns:
+        - node.infra*
+        settings:
+          number_of_shards: 1,        #value based on es node count policy
+          number_of_replicas: 1,      #value based on redundancy policy
+        aliases:
+        - infra  
+    seedings:
+    - prefix: node.infra              #creates node.infra-00001 aliased node.infra-write     
+      policyRef: infra-policy
+```
+#### Elasticsearch Operator
+The `elasticsearch-operator` will be modified to:
+* Expose the index management API
+* Create and seed specified index templates
+* Create the initial indices as needed
+* Block data ingestion if needed until initial index is seeded
+* Deploy curation CronJob for each entry in the specified seedings to rollover using the defined policy
+
+#### Collector 
+* Update the [`fluent-plugin-viaq-data-model`](https://github.com/ViaQ/fluent-plugin-viaq_data_model) to allow defining a static index to write logs
+
+Example configuration:
+```    
+  ....
+   <elasticsearch_index_name>
+     tag "**"
+     name_type static index_name=>'container.app-write'
+   </elasticsearch_index_name>
+```
+#### Deprecations
+* The Curator Cronjob deployed by the `cluster-logging-operator` will be removed. The responsibilities for curation will be subsumed by
+ implementation of Elasticsearch rollover management.
+* Curation configuration by namespace is no longer configurable and is restricted to cluster
+wide settings associated with log type 
+
+## Design Details
+
+### Test Plan
+* Regression tests will be executed to confirm no regressions from previous releases
+#### Unit Tests
+* Unit tests will be modified to account for the change in data design
+#### Integration Tests
+* e2e tests will be modified to account for the change in data design
+
+### Upgrade / Downgrade Strategy
+
+#### Upgrade
+The `elasticsearch-operator` will migrate existing indices to the new data design:
+* Indices beginning with `project.*` are indexed to `container.app-write`
+* Indices beginning with `.operations.*` are indexed to `journal.infra-write`
```

Comment:
is this because we currently co-locate our node and container infra logs? e.g. we just are trying to prevent losing logs with this (given we will be writing to a `node.` or `container.` for infra)

Reply:
Correct so we do not lose existing logs. It's possible we could actually spin through the `.operations` index and split out the journal and container logs but I imagine it is easier to move the index whole sale instead of trying to split it.   Opinions?

Reply:
Why have the distinction of container vs journal logs to start?

Reply:
Only reason it exists because node logs come from journal.  My original thought was to call it `node.XXXX`... would that be better?

Reply:
Perhaps so that it is not tied to a name referring to the implementation of how logs from the node are collected.

Reply:
replaced 'journal' with 'node'